### PR TITLE
Add AWS S3 integration test, add minio tests

### PR DESCRIPTION
**Pull request by ThomasMarwitzQC**
Extend the minio docker setup with a custom Dockerfile that uses mc
to configure 2 users with access to a different bucket each.

Fix handling of url param "verify" that can be passed to [h]s3 stores.

Add AWS integration tests and configure CI to authenticate to AWS.

### CODE REVIEW

**Comment by ThomasMarwitzQC** [comment]
TODO: Rebase some changes if this S3 integration is working as expected from `minimalkv` branch

File: tests/minio-container/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM minio/minio
+COPY scripts/startup.sh /startup.sh
+COPY policies /policies
+COPY files /files
+RUN chmod +x /startup.sh
```
Comment (xhochy) [inline tests/minio-container/Dockerfile:5]: This should not be necessary if you would checkin the file already with the executable bit set.
```suggestion
```

**Comment by ThomasMarwitzQC** [comment]
The checks look a bit weird bc apparently gh is a bit confused about the branch from the fork and the branch on the repo

**Comment by ThomasMarwitzQC** [comment]
Integration passed on minimalkv branch

**Review by xhochy — APPROVED** [review]

**Pull request merged**